app/assets/src/components/views/SampleViewV2/ReportTableV2.jsx

```diff
@@ -355,6 +288,67 @@ class ReportTableV2 extends React.Component {
     return defaultTableRowRenderer(rowProps);
   };
 
+  getNtNrFromDataRow(rowData, field, defaultValue) {
+    return [
+      getOr(defaultValue, ["nt", field], rowData),
+      getOr(defaultValue, ["nr", field], rowData),
+    ];
+  }
+
+  nestedSortFunction = ({ data, path, sortDirection, nullValue, limits }) => {
+    // Uses lodash's orderBy function.
+    // It uses a triple sorting key that enables nested sorting of genus and species, while guaranteeing that
+    // genus is always on top of its children species
+    return orderBy(
+      [
+        // 1st value: value defined by path for the genus (guarantees all genus together)
+        // note: a species row as a point .genus to their genus
+        rowData =>
+          rowData.genus
+            ? getOr(nullValue, ["genus"].concat(path), rowData)
+            : getOr(nullValue, path, rowData),
+        // 2nd value: the genus tax id
+        // this value guarantees that we keep species within their genus, even if the first value is duplicated
+        // e.g. if two genus have 2 reads they would be together on top and they species after them,
+        // adding tax id guarantees all the species are below their respective genus
```

Comment:
This comment was super helpful. Thanks for that. I definitely didn't understand all the nuances before.